src/utils/date_utils.py

```diff
@@ -20,6 +20,22 @@ def get_years_difference_from_current_date(utc_hour_offset: str, date: str, date
 def is_current_date(date: str, date_format: str = '%Y-%m-%d') -> bool:
     if not date:
         return False
+
+    if not is_valid_format_date(date, date_format):
+        return False
+
```

Comment:
Maybe combine both validations in the same if like `if not date or not is_valid_format_date() return False`

And maybe rename the function to -> `is_valid_date_format()`